lib/bolt/util.rb

```diff
@@ -78,11 +78,25 @@ def read_optional_yaml_hash(path, file_name)
       end
 
       def first_runs_free
+        # If this fails, use the system path instead
+        FileUtils.mkdir_p(Bolt::Config.user_path)
         Bolt::Config.user_path + '.first_runs_free'
+      rescue StandardError
+        begin
+          # If using the system path fails, then don't bother with the welcome
+          # message
+          FileUtils.mkdir_p(Bolt::Config.system_path)
+          Bolt::Config.system_path + '.first_runs_free'
+        rescue StandardError
+          nil
+        end
       end
 
       def first_run?
-        Bolt::Config.user_path && !File.exist?(first_runs_free)
+        # If the first_run file can be written, and hasn't already been written
+        !first_runs_free.nil? &&
+          File.writable?(first_runs_free) &&
```

Comment:
`File.writable?` checks whether the file has the `write` permission, not whether it could be created. It should be impossible for a file to both not exist but also be writable.